### DISCUSSION TITLE
Model.__fields__ in v2

### DISCUSSION THREAD

**Discussion by vitalik** [pydantic/pydantic#5909] [Pydantic V2]
Hi 

pydantic-v1 has this `__fields__` - a dictionary with all fields definded for that model:

```
>>> Data.__fields__
{'f1': ModelField(name='f1', type=int, required=True)}
```

what's the alternative in v2 ?

**Comment by vitalik** [comment]
`model_fields` it is  :)

**Comment by samuelcolvin** [comment]
Compatibility shim added in #5912.
